src/utils/pagedIterator.ts

```diff
@@ -0,0 +1,85 @@
+import { Issue as IssueRequest, Vault, Redeem, ActiveStakedRelayer } from "../interfaces/default";
+import { AugmentedQuery, PaginationOptions } from "@polkadot/api/types";
+import { StorageKey } from "@polkadot/types/primitive";
+import { Codec, Observable, CodecArg } from "@polkadot/types/types";
+import { ApiPromise } from "@polkadot/api";
+import { QueryableStorageEntry, ApiTypes } from "@polkadot/api/types";
+
+export type PolkadotCodecType = IssueRequest | Redeem | Vault | ActiveStakedRelayer;
+
+export async function* pagedIterator<T extends PolkadotCodecType>(
+    api: ApiPromise,
+    polkadotListings: AugmentedQuery<ApiTypes, (arg: CodecArg) => Observable<T>> & QueryableStorageEntry<ApiTypes>,
+    perPage: number
+): AsyncGenerator<T[]> {
+    let startKey: StorageKey = api.createType("Bytes", "") as StorageKey;
+    let issueRequests: [StorageKey, Codec][];
+    while (true) {
+        // a blank StorageKey is 0x in hex
+        if (startKey.toHex() == "0x") {
+            if (!isRemainingListNonempty(polkadotListings)) {
+                return undefined;
+            }
+            issueRequests = await listPaged(polkadotListings, perPage);
+        } else {
+            if (!isRemainingListNonempty(polkadotListings, startKey)) {
+                return undefined;
+            }
+            const keys = await api.query.issue.issueRequests.keysPaged(
+                { pageSize: perPage, startKey: startKey.toHex() } as PaginationOptions
+            );
+            if (keys.length) {
+                issueRequests = await listPaged(polkadotListings, perPage, startKey);
+            } else {
+                return undefined;
+            }
+        }
+
+        startKey = issueRequests[issueRequests.length - 1][0];
+        yield issueRequests.map((v) => v[1] as T);
+    }
+}
+
+async function listPaged<T extends PolkadotCodecType>(
+    // eslint-disable-next-line max-len
+    polkadotListings: AugmentedQuery<ApiTypes, (arg: CodecArg) => Observable<T>> & QueryableStorageEntry<ApiTypes>,
+    perPage: number,
+    startKey?: StorageKey
+): Promise<[StorageKey, Codec][]> {
+    let issueRequests: [StorageKey, Codec][] = [];
+
+    if (startKey) {
```

Comment:
This can be simplified quite a bit:

```typescript
const start = startKey ? startKey.toHex() : undefined;
return await polkadotListings.entriesPaged(
                { pageSize: perPage, startKey: start } as PaginationOptions
            ) as [StorageKey, Codec][];
```